File: packages/3/ms-3-expressjs/src/routes/country.routes.ts
import { iresponse, createOneCountrySchema, updateOneCountrySchema, deleteOneCountrySchema, searchOneCountrySchema } from '@gmahechas/erp-common';
import { express, validatorMiddleware } from '@gmahechas/erp-common-expressjs';
import { createOneCountry, updateOneCountry, deleteOneCountry, searchOneCountry } from '@gmahechas/erp-common-ms-3-js';

const router = express.Router();

router.post('/create/one', validatorMiddleware(createOneCountrySchema), async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await createOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.put('/update/one', validatorMiddleware(updateOneCountrySchema), async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await updateOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.delete('/delete/one', validatorMiddleware(deleteOneCountrySchema), async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await deleteOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.post('/search/one', validatorMiddleware(searchOneCountrySchema), async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await searchOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

export default router;

File: packages/3/ms-3-flaskpy/flaskr/routes/v1/country_routes.py
from gmahechas_erp_common_py import response
from gmahechas_erp_common_flaskpy import Blueprint, request, Response, authentication_middleware
from gmahechas_erp_common_ms_3_py import CountryController

country_blueprint = Blueprint('country', __name__, url_prefix='/country')


@country_blueprint.route('/create/one', methods=['POST'])
@authentication_middleware
def create_one():
    country = CountryController()
    data = response(200, country.create_one_country(request.json))
    status_code, body = data['statusCode'], data['body']
    return Response(body, mimetype='application/json', status=status_code)


@country_blueprint.route('/update/one', methods=['PUT'])
@authentication_middleware
def update_one():
    country = CountryController()
    data = response(200, country.update_one_country(request.json))
    status_code, body = data['statusCode'], data['body']
    return Response(body, mimetype='application/json', status=status_code)


@country_blueprint.route('/delete/one', methods=['DELETE'])
@authentication_middleware
def delete_one():
    country = CountryController()
    data = response(200, country.delete_one_country(request.json))
    status_code, body = data['statusCode'], data['body']
    return Response(body, mimetype='application/json', status=status_code)


@country_blueprint.route('/search/one', methods=['POST'])
@authentication_middleware
def search_one():
    country = CountryController()
    data = response(200, country.search_one_country(request.json))
    status_code, body = data['statusCode'], data['body']
    return Response(body, mimetype='application/json', status=status_code)


@country_blueprint.route('/search/many', methods=['POST'])
@authentication_middleware
def search_many():
    country = CountryController()
    data = response(200, country.search_many_country(request.json))
    status_code, body = data['statusCode'], data['body']
    return Response(body, mimetype='application/json', status=status_code)
